src/liballoc/Cargo.toml

```diff
@@ -11,6 +11,9 @@ path = "lib.rs"
 core = { path = "../libcore" }
 std_unicode = { path = "../libstd_unicode" }
 
+[target.'cfg(any(unix, target_os = "redox"))'.dependencies]
+libc = { path = "../rustc/libc_shim" }
```

Comment:
The `alloc` crate has actually up to this point pretty intentionally avoided a dependency on `libc` (aka namely a dependency on any assumptions about the OS). It's intended that `alloc` *purely* assumes the existence of a global allocator but continues to assume nothing else.

I think the dependency here is for `oom`, right? Is there perhaps a way we could work around that an keep the `oom` definition in alloc_system/std?

Reply:
One possible idea could be to add an `oom` method to the `GlobalAlloc` trait which defaults to `intrinsics::abort`, but we could override that in `alloc_system` or something like that